Utilities/StorageFactory/src/RemoteFile.cc

```diff
@@ -53,27 +54,27 @@ int RemoteFile::local(const std::string &tmpdir, std::string &temp) {
   // likely to have more space, and is more optimised for
   // large files, and is cleaned up after the job.
   if (tmpdir.empty() || tmpdir == ".") {
-    size_t len = pathconf(".", _PC_PATH_MAX);
-    char *buf = (char *)malloc(len);
-    getcwd(buf, len);
-
-    temp.reserve(len + 32);
-    temp = buf;
-    free(buf);
+    fs::path current_path = fs::current_path();
+    temp = current_path.string();
   } else {
-    temp.reserve(tmpdir.size() + 32);
```

Comment:
I think using temp here and then temp_chars is not a good way to go. My suggestion is to just declare `temp_chars` early and copy all the info from `current_path` and `storage-factory-local-XXXXXX` directly into `temp_chars`, use `temp_chars` in `mkstemp` and only when done, copy `temp_chars` to `temp`.